### ISSUE TITLE
Full inventory no longer triggers Lab Fault

### ISSUE THREAD

**Issue by slartifer** [HughJeffner/FFRK-LabMem#179]
This is pretty minor, but a full inventory used to trigger a lab fault, which was useful for notification purposes.  Now it doesn't, you just get endless restarts instead -- not sure if something changed on DeNA's end.

**Comment by HughJeffner** [comment]
From my testing it still appears to work

I did notice if the tap on the first 'enter' button doesn't register which sometimes happens (lag?) then it gets confused and would get stuck.  This doesn't explain your restart loop however.

**Comment by slartifer** [comment]
Weird.  This has happened to me twice recently.  It opens the corridor successfully and ends up in a restart loop on the first battle it encounters.  Didn't have debug info on at the time, but I'll see if I can catch it again.

**Comment by HughJeffner** [comment]
This is after you enter the lab?  I only tested the inventory full dialog (with all the sell/shatter options) when trying to enter the lab.

I know there was a separate issue where you go over inventory while in a forced battle #119 

I suppose you can pause a lab run, go over inventory, then resume it and it would get stuck in a loop in the next battle

**Comment by slartifer** [comment]
Okay, I _thought_ I was previously getting that error on entering lab, but wasn't sure if I was crazy!  Maybe DeNA removed the on-entry inventory limit check?  In both cases this happened on its own during a sequence of lab runs, and was triggered by the first battle rather than the lab entry itself.  There was no manual operation, therefore no chance for the inventory to change in between lab entry and the first battle.

**Comment by HughJeffner** [comment]
I tested it by creating a bunch of rarity 1 fire abilities, maybe over HE inventory doesn't trigger the dialog?  Will need to test that.

**Comment by slartifer** [comment]
fire! fire! fire! fire! huh huh huh huh

HE inventory had to be what triggered this for me, since it happened in the course of sequential lab runs.  But I'm quite certain this is _also_ what's successfully triggered lab faults for me in the past.  Is it possible it's just tapping too fast to recognize the inventory full dialog?  Or maybe it's backing out of that in an attempt to recover?

**Comment by HughJeffner** [comment]
There's only one place where it checks for inventory full and that's when starting a lab

They might have changed the entry to not check for HE?  Not sure.  Gonna test it when my inventory gets full enough

**Comment by slartifer** [comment]
Here are the (simple) logs from one of the times it happened:

```
08:21:32 We reached the master painting.  
08:21:32 Using [Teleport Stone] x1
08:21:39 Lab run completed! (1:23:23)
08:21:39 Restarting Lab in 60 seconds...
08:22:08 Restarting Lab in 30 seconds...
08:22:28 Restarting Lab in 10 seconds...
08:22:39 Restarting Lab
08:22:58 Current lab set to: Corridor of the Dragon King S3
08:23:00 Welcome to Floor 1!
08:23:02 Picking painting 2: Combatant Painting: Titan (Labyrinth) [R]
08:23:05 Battle Info
08:23:35 Possible hang, attempting recovery!
08:23:35 Navigating back
08:23:40 Trying to auto-start
08:23:41 Could not auto-start
08:24:35 Hang detected!
08:24:35 Restarting FFRK
08:24:59 Picking painting 2: Combatant Painting: Titan (Labyrinth) [R]
08:25:01 FFRK restarted!
08:25:02 Battle Info
08:25:32 Possible hang, attempting recovery!
08:25:32 Navigating back
08:25:38 Trying to auto-start
08:25:39 Could not auto-start
08:26:32 Hang detected!
08:26:32 Restarting FFRK
08:26:56 Picking painting 2: Combatant Painting: Titan (Labyrinth) [R]
08:26:57 FFRK restarted!
08:26:59 Battle Info
08:27:29 Possible hang, attempting recovery!
08:27:29 Navigating back
08:27:34 Trying to auto-start
08:27:35 Could not auto-start
08:28:29 Hang detected!
08:28:29 Restarting FFRK
...
09:21:13 Picking painting 2: Combatant Painting: Titan (Labyrinth) [R]
09:21:13 FFRK restarted!
09:21:16 Battle Info
09:21:46 Possible hang, attempting recovery!
09:21:46 Navigating back
09:21:51 Trying to auto-start
09:21:52 Could not auto-start
09:22:46 Restart loop detected!
09:22:46 Disabled Lab
```

When I came back to the device about 10 minutes later, it was on the "inventory full" screen.  I don't think it fabricated the json for the paintings and Titan, so there's definitely a check just before battle that doesn't happen on corridor entry.

**Comment by HughJeffner** [comment]
Well I just spent ~900 treasure maps for science and it did indeed give me the screen when trying to enter the lab

Maybe it handles it differently when going over the limit after completing a lab vs outside the lab.

I'll go 1 HE under the limit and try a QE and see what it does.

**Comment by HughJeffner** [comment]
Alright I was finally able to reproduce this.  Doing a QE to go over the limit did *not* work as I got the dialog upon entry.  Also beating the master painting to go over the limit didn't allow me to enter the lab either.

What I had to do was get 1 HE under the limit and enter the lab, make sure to pick up at least 1 HE in the run, then exit out with a teleport stone.  The **teleport stone is the key** as the FFRK client doesn't seem to compare the HE count correctly after using it.

If the above conditions are met then it will let you in the lab and either get the system error on a forced battle (#119) or you finally get the inventory dialog when entering a combatant painting.  Since the bot does not expect this, we get the restart loop.
